tests/v2/integration/catalogv2/system_charts_version_test.go

```diff
@@ -0,0 +1,272 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	stevev1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/clients/rancher/catalog"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+)
+
+type SystemChartsVersionSuite struct {
+	suite.Suite
+	client               *rancher.Client
+	session              *session.Session
+	restClientGetter     genericclioptions.RESTClientGetter
+	catalogClient        *catalog.Client
+	latestWebhookVersion string
+}
+
+func (w *SystemChartsVersionSuite) TearDownSuite() {
+	_ = w.updateSetting("rancher-webhook-version", w.latestWebhookVersion)
+	_ = w.updateSetting("rancher-webhook-min-version", "")
+	_ = w.updateSetting("system-feature-chart-refresh-seconds", "3600")
+	w.session.Cleanup()
+}
+
+func (w *SystemChartsVersionSuite) SetupSuite() {
+	var err error
+	testSession := session.NewSession()
+	w.session = testSession
+
+	w.client, err = rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+
+	w.catalogClient, err = w.client.GetClusterCatalogClient("local")
+	require.NoError(w.T(), err)
+
+	host := hostName(w.client.RestConfig.Host)
+	w.restClientGetter = &genericclioptions.ConfigFlags{
```

Comment:
Max instead of exporting the rest.Config we have extensions that do this.
  ```
kubeConfig, err := kubeconfig.GetKubeconfig(client, clusterID)
  require.NoError()
  restConfig, err := (*kubeConfig).ClientConfig()
  require.NoError()
 
 newClientGett := NewRestGetter(restConfig, clientConfig)
```
NewRestGetteer is in https://github.com/rancher/rancher/blob/release/v2.7/tests/framework/extensions/kubeconfig/restgetter.go

Reply:
Thanks for the link! For this to work, I might need some help implementing the `ToDiscoveryClient` method on the `RestGetter`. It gets called by Helm, and our getter just returns nil for that now.

Reply:
Let me know!

Reply:
Hey Max I know this needs to be merged, but to avoid making a pretty fundamental change to the framework could you get the restGetter to work by get the restConfig like this instead?
```
kubeConfig, err := kubeconfig.GetKubeconfig(client, "local")
require.NoError()
restConfig, err := (*kubeConfig).ClientConfig()
require.NoError()

```

Reply:
Yup, it's done, will open a follow-up PR soon.